packages/web/components/bridge/amount-screen.tsx

```diff
@@ -351,23 +356,36 @@ export const AmountScreen = observer(
 
     const { data: assetsBalances, isLoading: isLoadingAssetsBalance } =
       api.local.bridgeTransfer.getSupportedAssetsBalances.useQuery(
-        fromChain?.chainType === "evm"
-          ? {
-              type: "evm",
-              assets: supportedSourceAssets as Extract<
-                SupportedAsset,
-                { chainType: "evm" }
-              >[],
-              userEvmAddress: evmAddress,
-            }
-          : {
-              type: "cosmos",
-              assets: supportedSourceAssets as Extract<
-                SupportedAsset,
-                { chainType: "cosmos" }
-              >[],
-              userCosmosAddress: fromCosmosCounterpartyAccount?.address,
-            },
+        {
+          source:
+            fromChain?.chainType === "evm"
+              ? {
+                  type: "evm",
+                  assets: supportedSourceAssets as Extract<
+                    SupportedAsset,
+                    { chainType: "evm" }
+                  >[],
+                  userEvmAddress: evmAddress,
+                }
+              : fromChain?.chainType === "cosmos"
+              ? {
+                  type: "cosmos",
+                  assets: supportedSourceAssets as Extract<
+                    SupportedAsset,
+                    { chainType: "cosmos" }
+                  >[],
+                  userCosmosAddress: fromCosmosCounterpartyAccount?.address,
+                }
+              : {
+                  // Note: for now, these will always return 0 balances since bitcoin and solana are not supported
+                  type: fromChain?.chainType ?? "bitcoin",
+                  assets:
+                    (supportedSourceAssets as Extract<
+                      SupportedAsset,
+                      { chainType: "bitcoin" }
+                    >) ?? [],
+                },
+        },
```

Comment:
Thoughts on using a self-calling function to simplify the conditional flow so it's more readable? 

```suggestion
const assetDetails = (() => {
  if (fromChain?.chainType === "evm") {
    return {
      type: "evm",
      assets: supportedSourceAssets.filter(asset => asset.chainType === "evm"),
      userEvmAddress: evmAddress,
    };
  } else if (fromChain?.chainType === "cosmos") {
    return {
      type: "cosmos",
      assets: supportedSourceAssets.filter(asset => asset.chainType === "cosmos"),
      userCosmosAddress: fromCosmosCounterpartyAccount?.address,
    };
  } else {
    return {
      type: fromChain?.chainType ?? "bitcoin",
      assets: supportedSourceAssets.filter(asset => asset.chainType === "bitcoin") ?? [],
    };
  }
})();
```